File: src/main/java/com/zerozero/store/domain/service/ReadUserStoresUseCase.java
package com.zerozero.store.domain.service;

import com.zerozero.store.domain.repository.StoreRepository;
import com.zerozero.store.domain.response.StoreResponse;
import com.zerozero.user.domain.model.User;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;
import java.util.stream.Collectors;

@Service
@RequiredArgsConstructor
@Transactional(readOnly = true)
public class ReadUserStoresUseCase {

    private final StoreRepository storeRepository;

    public List<StoreResponse> execute(User user) {
        return storeRepository.findAllByUserIdWithImages(user.getId())
                .stream()
                .map(StoreResponse::from)
                .collect(Collectors.toList());
    }

}

File: src/main/java/com/zerozero/store/domain/service/ReadUserStoresUseCase.kt
package com.zerozero.store.domain.service

import com.zerozero.store.domain.repository.StoreRepository
import com.zerozero.store.domain.response.StoreResponse
import com.zerozero.user.domain.model.User
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
@Transactional(readOnly = true)
class ReadUserStoresUseCase(
    private val storeRepository: StoreRepository
) {

    fun execute(user: User): List<StoreResponse> {
        return storeRepository.findAllByUserIdWithImages(user.id).map { StoreResponse.from(it) }
    }
}
